crates/rspack_plugin_javascript/src/visitors/dependency/scanner.rs

```diff
@@ -123,30 +123,55 @@ impl DependencyScanner<'_> {
     if let Callee::Import(_) = node.callee {
       if let Some(dyn_imported) = node.args.get(0) {
         if dyn_imported.spread.is_none() {
-          if let Expr::Lit(Lit::Str(imported)) = dyn_imported.expr.as_ref() {
-            let chunk_name = self.try_extract_webpack_chunk_name(&imported.span);
-            self.add_dependency(box EsmDynamicImportDependency::new(
-              imported.value.clone(),
-              Some(node.span.into()),
-              as_parent_path(ast_path),
-              chunk_name,
-            ));
-          }
-          if let Some((context, reg)) = scanner_context_module(dyn_imported.expr.as_ref()) {
-            self.add_dependency(box ImportContextDependency::new(
-              ContextOptions {
-                mode: ContextMode::Lazy,
-                recursive: true,
-                reg_exp: RspackRegex::new(&reg).expect("reg failed"),
-                reg_str: reg,
-                include: None,
-                exclude: None,
-                category: DependencyCategory::Esm,
-                request: context,
-              },
-              Some(node.span.into()),
-              as_parent_path(ast_path),
-            ));
+          match dyn_imported.expr.as_ref() {
+            Expr::Lit(Lit::Str(imported)) => {
+              let chunk_name = self.try_extract_webpack_chunk_name(&imported.span);
+              self.add_dependency(box EsmDynamicImportDependency::new(
+                imported.value.clone(),
+                Some(node.span.into()),
+                as_parent_path(ast_path),
+                chunk_name,
+              ));
+            }
+            Expr::Tpl(tpl) => match Some(&tpl.quasis) {
```

Comment:
I might cause some misunderstanding to you :)

I'm suggesting changes like this
```rust
match dyn_imported.expr.as_ref() {
   Expr::Tpl( { quasis }) if quasis.len() == 1 => {
     // ..
   },
   Expr::Tpl(Tpl{ quasis, .. }) if quasis.len() > 1 => {
     // ..
   },
}
```

- You don't need `Option` to use pattern matching with guard
- `Tpl` in `Tpl{ quasis, .. }` is imported from `swc_core::ecma::ast::Tpl`